src/core/include/openvino/op/transpose.hpp

```diff
@@ -35,8 +35,37 @@ class OPENVINO_API Transpose : public Op {
 
     OPENVINO_SUPPRESS_DEPRECATED_START
     bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
+    bool evaluate_upper(const HostTensorVector& output_values) const override;
+    bool evaluate_lower(const HostTensorVector& output_values) const override;
     OPENVINO_SUPPRESS_DEPRECATED_END
+
     bool has_evaluate() const override;
+    bool evaluate_label(TensorLabelVector& output_labels) const override;
+
+    /// \brief Generates default axes order at end of input vector.
+    ///
+    /// Default axes order is decreasing sequence numbers which start from `length - 1`.
+    ///
+    /// \param axes_order  Vector where default order will be generated.
+    /// \param length      Sequence length of axes order.
+    ///
+    static void generate_default_order(std::vector<int64_t>& axes_order, const size_t length);
+
+    /// \brief Check if vector of axes order has got valid values.
+    ///
+    /// Axes order has to be unique numbers in range of [0, size)
+    ///
+    /// \param axes_order  Vector with axes order to check.
+    /// \param size        Input for transpose rank size.
+    ///
+    /// \return true if axes order is valid otherwise false.
+    ///
+    static bool is_valid_order(const std::vector<int64_t>& axes_order, const size_t size);
```

Comment:
This is not the best place for these utility methods. I would love to keep operation class clean. We have validation_util file for helpers. @ilyachur do you agree?
1 -- backward compatibility reasons
2 -- binary size should be minimal for ops

Reply:
If required can move it to separate file. I've seen that some others operator have got some custom methods.
Just not sure if `validation_util` is good place as I think it should have some generic utils.
Can create some dedicated files in same place as `validation_util` which will be more transpose specific.

Reply:
Moved to `validatoin_util`